### PR TITLE
add cephfs rdr cg design doc

### PR DESCRIPTION
**Pull request by youhangwang**

### CODE REVIEW

File: docs/design/Cephfs-RDR-ConsistencyGroup.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+Ramen takes on the responsibility of creating volume group snapshots for
+the PVCs within the consistency group and restoring these snapshots to PVCs.
+Subsequently, Volsync synchronizes the data from the PVCs restored by
+Ramen to the remote destination.
+
+So, in step 10, there are some differences compared to the
+current implementation if a consistency group is specified:
+
+* Ramen get the existing VolumeGroupSnapshotClass
+* Ramen creates a ReplicationGroupSource with the schedule
```
Comment (BenamarMk) [inline docs/design/Cephfs-RDR-ConsistencyGroup.md:273]: Ramen will have a scheduler (replacing VolSync scheduler) to manage the scheduling of ReplicationGroupSnapshot requests at schedule intervals. Please ensure that the doc reflects this change.

File: docs/design/Cephfs-RDR-ConsistencyGroup.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+}
+
+type ReplicationGroupSourceSpec struct {
+  // Trigger to do the schedule sync or final sync
+  Trigger *ReplicationSourceTriggerSpec `json:"trigger,omitempty"`
+
+  VolumeGroupSnapshotClassName *string `json:"volumeGroupSnapshotClassName,omitempty"`
+
+  // RSSpecs specify the PVCs in a volume group
+  RSSpecs []VolSyncReplicationSourceSpec `json:"rsspecs,omitempty"`
```
Comment (BenamarMk) [inline docs/design/Cephfs-RDR-ConsistencyGroup.md:87]: This should contain ReplicationSources

File: docs/design/Cephfs-RDR-ConsistencyGroup.md [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec   ReplicationGroupDestinationSpec `json:"spec,omitempty"`
+  Status *ReplicationDestinationStatus   `json:"status,omitempty"`
+}
+
+type ReplicationGroupDestinationSpec struct {
+  // RDSpecs specify the PVCs in a volume group
+  RDSpecs []VolSyncReplicationDestinationSpec `json:"rdspecs,omitempty"`
```
Comment (BenamarMk) [inline docs/design/Cephfs-RDR-ConsistencyGroup.md:129]: This should contain ReplicationDestinations

File: docs/design/Cephfs-RDR-ConsistencyGroup.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+        * Each snapshot is then restored to a new read-only PVC by Ramen
+        * Ramen creates a ReplicationSource (Use Manual and Direct
+          method in ReplicationSource) for each restored PVC
+        * Ramen monitors the manual string in the ReplicationSource
+          spec and status to ensure all PVCs in the group are
+          successfully synced to the remote destination.
+        * Once synchronization is completed, update Status.LastSyncStartTime
+          to nil, and Status.LastSyncTime. Go to Cleanup state
+    * In Cleanup state:
+        * Ramen cleans up each RS and the read-only PVCs
```
Comment (BenamarMk) [inline docs/design/Cephfs-RDR-ConsistencyGroup.md:207]: Should not cleanup each RS. Each RS should stay around maintaining its status information until the next successful sync. In other words, once the RS for a PVC is created, it should stay around forever, just the sourcePVC and the trigger string should change with every new sync.

**Review by BenamarMk — COMMENTED** [review]
The design is pretty much in a good state.  I added just a few comments.

File: docs/design/Cephfs-RDR-ConsistencyGroup.md (hunk 2 above)
Comment (youhangwang) [reply docs/design/Cephfs-RDR-ConsistencyGroup.md:87]: Add the created ReplicationSources in the status.
- Spec.RSSpecs is the protectedPVCs show how to create the ReplicationSources
- Status.ReplicationSources show the created ReplicationSources reference

File: docs/design/Cephfs-RDR-ConsistencyGroup.md (hunk 3 above)
Comment (youhangwang) [reply docs/design/Cephfs-RDR-ConsistencyGroup.md:129]: same with ReplicationSources

File: docs/design/Cephfs-RDR-ConsistencyGroup.md [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+  NextSyncTime *metav1.Time `json:"nextSyncTime,omitempty"`
+  // lastManualSync is set to the last spec.trigger.manual when the manual sync is done.
+  //+optional
+  LastManualSync string `json:"lastManualSync,omitempty"`
+  // conditions represent the latest available observations of the
+  // source's state.
+  Conditions []metav1.Condition `json:"conditions,omitempty"`
+  
+  // Created ReplicationSources by this ReplicationGroupSource
+  ReplicationSources []*corev1.TypedLocalObjectReference `json:"replicationSources,omitempty"`
```
Comment (BenamarMk) [inline docs/design/Cephfs-RDR-ConsistencyGroup.md:114]: Maybe change `TypedLocalObjectReference` to be `ObjectReference` instead. The `ReplicationGroupSource` may contain PVCs from different namespaces.

File: docs/design/Cephfs-RDR-ConsistencyGroup.md [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+  LastSyncDuration *metav1.Duration `json:"lastSyncDuration,omitempty"`
+  // latestImage in the objects holding the most recent consistent replicated
+  // image.
+  //+optional
+  LatestImages []*corev1.TypedLocalObjectReference `json:"latestImage,omitempty"`
+
+  Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+  // Created ReplicationDestinations by this ReplicationGroupDestination
+  ReplicationDestinations []*corev1.TypedLocalObjectReference `json:"replicationDestinations,omitempty"`
```
Comment (BenamarMk) [inline docs/design/Cephfs-RDR-ConsistencyGroup.md:154]: Same comment as ReplicationSources above

**Review by BenamarMk — COMMENTED** [review]
Looks good to me.

**Review by BenamarMk — APPROVED** [review]

**Pull request merged**